src/Core/ClassInfo.php

```diff
@@ -301,14 +301,14 @@ public static function classImplements($className, $interfaceName)
      */
     public static function classes_for_file($filePath)
     {
-        $absFilePath = Director::getAbsFile($filePath);
+        $absFilePath = Director::getAbsFile(realpath($filePath));
         $classManifest = ClassLoader::inst()->getManifest();
         $classes = $classManifest->getClasses();
         $classNames = $classManifest->getClassNames();
 
         $matchedClasses = [];
         foreach ($classes as $lowerClass => $compareFilePath) {
-            if (strcasecmp($absFilePath ?? '', $compareFilePath ?? '') === 0) {
+            if (strcasecmp($absFilePath ?? '', realpath($compareFilePath) ?? '') === 0) {
```

Comment:
Is this really needed for the feature? I'd feel better if we didn't have to touch unrelated parts.

Reply:
I am running this on Windows and the paths don't match if they are not cleaned up. I can submit a separate PR for this if you prefer?

Reply:
This should be a separate pr please.

Reply:
done, see https://github.com/silverstripe/silverstripe-framework/pull/10687